### PR TITLE
U/jrbogart/trim bandpass

### PR DESCRIPTION
**Pull request by JoanneBogart**
Includes
* new code to thin and cache lsst bandpasses
* when creating flux catalogs, if --flux-parallel option value is 1, do processing inline rather than forking subprocesses 
*  bug fix to make use of flux files efficient as intended
* bug fix to keep going if in `get_objects_by_region` not all healpixel files intersecting the region are present

### CODE REVIEW

File: python/desc/skycatalogs/skyCatalogs.py [hunk 1]
```diff
@@ -359,6 +360,9 @@ def get_objects_by_hp(self, hp, region=None, obj_type_set=None,
         '''
         object_list = ObjectList()
 
+        if hp not in self._hp_info:
```
Comment (jchiang87) [inline python/desc/skycatalogs/skyCatalogs.py:363]: I think you had mentioned something about putting a warning here if a healpixel intersecting the requested region didn't have a corresponding catalog file.  I agree that might be useful info to output here.

File: python/desc/skycatalogs/catalog_creator.py [hunk 2]
```diff
@@ -430,7 +433,7 @@ def create_galaxy_flux_catalog(self, config_file=None):
         from desc.skycatalogs import open_catalog, SkyCatalog
 
         self._gal_flux_schema = make_galaxy_flux_schema(self._logname)
-        BaseObject.get_bp_dir()
+        ###BaseObject.get_bp_dir()
```
Comment (jchiang87) [inline python/desc/skycatalogs/catalog_creator.py:436]: Any reason to not just delete this line?

**Review by jchiang87 — APPROVED** [review]
Looks good to me.  I do have one feature request to add to this PR if you feel so inclined:  Can you provide access to the `verbose` flag in the `open_catalog` function and pass that value to the `SkyCatalog` object on creation?

File: python/desc/skycatalogs/skyCatalogs.py (hunk 1 above)
Comment (JoanneBogart) [reply python/desc/skycatalogs/skyCatalogs.py:363]: Ok, I'll add that.

**Comment by JoanneBogart** [comment]
Concerning access to the verbose flag - easy enough to do, but I'm not sure how useful it is. I'll scan the code to see what gets output when it's set.   I had been thinking of getting rid of output from the verbose command in favor of logging messages at DEBUG level, but I don't yet use the logger consistently.

**Comment by jchiang87** [comment]
If you want to use the logger instead that's fine, but I did find it useful to be able to set that verbose flag through the `open_catalog` function in my copy of the repo.   I think as long as there are print statements controlled by that verbose flag, I would find it convenient to be able to control that output via that option through `open_catalog`.

**Comment by JoanneBogart** [comment]
Since you found it useful I'll do that now; it's quick and easy.  Ultimately I think using the logger is better but that can wait.

File: python/desc/skycatalogs/catalog_creator.py (hunk 2 above)
Comment (JoanneBogart) [reply python/desc/skycatalogs/catalog_creator.py]: Nope.  There were a bunch more like that that I did delete.  Fixed in new commit.

**Comment by JoanneBogart** [comment]
@jchiang87 do you want to look at this again or should I just go ahead and merge?

**Pull request merged**